File: code/java/rest/server/src/main/java/com/taskapi/model/Task.java
package com.taskapi.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {
    private String id;
    
    @NotBlank(message = "Title is required")
    @Size(max = 200, message = "Title must be 200 characters or less")
    private String title;
    
    private String description;
    
    @Builder.Default
    private TaskStatus status = TaskStatus.PENDING;
    
    @Builder.Default
    private TaskPriority priority = TaskPriority.MEDIUM;
    
    private List<String> tags;
    
    private String assignedTo;
    
    private LocalDateTime dueDate;
    
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    private String createdBy;
    
    private String updatedBy;
    
    public static Task create() {
        return Task.builder()
                .id(UUID.randomUUID().toString())
                .createdAt(LocalDateTime.now())
                .updatedAt(LocalDateTime.now())
                .build();
    }
}

File: code/kotlin/rest/server/src/main/kotlin/com/taskapi/model/Task.kt
package com.taskapi.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
enum class TaskStatus {
    PENDING, IN_PROGRESS, COMPLETED, CANCELLED, ARCHIVED
}

@Serializable
enum class TaskPriority {
    LOW, MEDIUM, HIGH, CRITICAL
}

@Serializable
data class Task(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val description: String? = null,
    val status: TaskStatus = TaskStatus.PENDING,
    val priority: TaskPriority = TaskPriority.MEDIUM,
    val tags: List<String> = emptyList(),
    val assignedTo: String? = null,
    val dueDate: Instant? = null,
    val createdAt: Instant,
    val updatedAt: Instant,
    val createdBy: String? = null,
    val updatedBy: String? = null
) {
    companion object {
        fun create(
            title: String,
            description: String? = null,
            priority: TaskPriority = TaskPriority.MEDIUM,
            tags: List<String> = emptyList(),
            assignedTo: String? = null,
            dueDate: Instant? = null
        ): Task {
            val now = kotlinx.datetime.Clock.System.now()
            return Task(
                title = title,
                description = description,
                priority = priority,
                tags = tags,
                assignedTo = assignedTo,
                dueDate = dueDate,
                createdAt = now,
                updatedAt = now
            )
        }
    }
}

@Serializable
data class CreateTaskRequest(
    val title: String,
    val description: String? = null,
    val priority: TaskPriority = TaskPriority.MEDIUM,
    val tags: List<String> = emptyList(),
    val assignedTo: String? = null,
    val dueDate: Instant? = null
)

@Serializable
data class UpdateTaskRequest(
    val title: String? = null,
    val description: String? = null,
    val status: TaskStatus? = null,
    val priority: TaskPriority? = null,
    val tags: List<String>? = null,
    val assignedTo: String? = null,
    val dueDate: Instant? = null
)

@Serializable
data class UpdateStatusRequest(
    val status: TaskStatus
)

@Serializable
data class TasksResponse(
    val items: List<Task>,
    val pageSize: Int,
    val nextPageToken: String? = null,
    val totalCount: Int
)

@Serializable
data class ErrorResponse(
    val error: String,
    val message: String
)

data class QueryParams(
    val pageSize: Int = 20,
    val pageToken: String? = null,
    val status: TaskStatus? = null,
    val assignedTo: String? = null,
    val tags: String? = null,
    val sortOrder: String? = null
)